lib/services/presets.go

```diff
@@ -316,6 +400,34 @@ func defaultAllowAccessReviewConditions(enterprise bool) map[string]*types.Acces
 func AddRoleDefaults(role types.Role) (types.Role, error) {
 	changed := false
 
+	// Role labels
+	defaultRoleLabels, ok := bootstrapRoleMetadataLabels()[role.GetName()]
+	if ok {
+		metadata := role.GetMetadata()
+
+		if metadata.Labels == nil {
+			metadata.Labels = make(map[string]string, len(defaultRoleLabels))
+		}
+		for label, value := range defaultRoleLabels {
+			if _, ok := metadata.Labels[label]; !ok {
+				metadata.Labels[label] = value
+				changed = true
+			}
+		}
+
+		if changed {
+			role.SetMetadata(metadata)
+		}
+	}
+
+	// Check if the role has a TeleportInternalResourceType attached. We do this after setting the role metadata
+	// labels because we set the role metadata labels for roles that have been well established (access,
+	// editor, auditor) that may not already have this label set, but we don't set it for newer roles
+	// (group-access, reviewer, requester) that may have customer definitions.
+	if role.GetMetadata().Labels[types.TeleportInternalResourceType] != types.PresetResource {
+		return nil, trace.AlreadyExists("not modifying user created role")
```

Comment:
Just making sure: will caller handle this error gracefully and not crash the startup?

Reply:
That's correct.